src/VisualStudio/Core/Def/ValueTracking/TreeViewItemBase.cs

```diff
@@ -33,6 +40,97 @@ public bool IsLoading
 
         public event PropertyChangedEventHandler? PropertyChanged;
 
+        public TreeViewItemBase()
+        {
+            ChildItems.CollectionChanged += (s, a) =>
+            {
+
+                if (a.Action is not NotifyCollectionChangedAction.Add and not NotifyCollectionChangedAction.Remove)
+                {
+                    return;
+                }
+
+                SetParents(a.OldItems, null);
+                SetParents(a.NewItems, this);
+
+                static void SetParents(IList? items, TreeViewItemBase? parent)
+                {
+                    if (items is null)
+                    {
+                        return;
+                    }
+
+                    foreach (var item in items.Cast<TreeViewItemBase>())
+                    {
+                        item.Parent = parent;
+                    }
+                }
+            };
```

Comment:
Call me old, but a 20 line lambda with a local function should probably just be a method 😛

Reply:
maybe... you don't always know when you start writing it :P